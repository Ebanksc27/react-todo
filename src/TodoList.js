import React, { useState } from 'react';
import NewTodoForm from './NewTodoForm';
import Todo from './Todo';

function TodoList() {
    const [todos, setTodos] = useState([]);
  
    const addTodo = newTodo => {
      setTodos(oldTodos => [...oldTodos, newTodo]);
    };
  
    const removeTodo = id => {
        setTodos(oldTodos => oldTodos.filter(todo => todo.id !== id));
      };
      
      return (
        <div>
          <NewTodoForm addTodo={addTodo} />
          {todos.map(todo => (
            <Todo key={todo.id} {...todo} removeTodo={removeTodo} />
          ))}
        </div>
      );
  }

export default TodoList;
