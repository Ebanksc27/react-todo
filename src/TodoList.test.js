import React from 'react';
import { render } from '@testing-library/react';
import TodoList from './TodoList';

test('renders TodoList component without crashing', () => {
  render(<TodoList />);
});

test('TodoList component snapshot', () => {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});
