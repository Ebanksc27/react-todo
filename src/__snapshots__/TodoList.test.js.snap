// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList component snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <input
        placeholder="New task"
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Add Todo
      </button>
    </form>
  </div>
</DocumentFragment>
`;
